### PR TITLE
Selection::fetchAssoc fix

### PR DESCRIPTION
**Pull request by matej21**
- http://forum.nette.org/cs/22696-fetchassoc-hazi-notice-undefined-index

### CODE REVIEW

**Comment by matej21** [comment]
Or should `Nette\Utils\Arrays::associate` use `iterator_to_array` for `Traversable` objects instead of [array conversion](https://github.com/nette/utils/blob/master/src/Utils/Arrays.php#L191)?

**Comment by dg** [comment]
Is it BC break?

File: src/Database/Table/Selection.php [hunk 1]
```diff
@@ -265,7 +265,9 @@ public function fetchAll()
 	 */
 	public function fetchAssoc($path)
 	{
-		return Nette\Utils\Arrays::associate($this->fetchAll(), $path);
+		return Nette\Utils\Arrays::associate(array_map(function(IRow $row) {
+			return iterator_to_array($row);
+		}, $this->fetchAll()), $path);
```
Comment (JanTvrdik) [inline src/Database/Table/Selection.php]:
~~~php
Nette\Utils\Arrays::associate(array_map('iterator_to_array', $this->fetchAll()))
~~~

**Comment by JanTvrdik** [comment]
BTW: I would personally prefer to solve this by somehow allowing `fetch()` to return raw array. In dibi you can for example do `$result->setRowClass(NULL)`.

File: src/Database/Table/Selection.php (hunk 1 above)
Comment (dg) [inline src/Database/Table/Selection.php]: iterator_to_array is slow, better is `(array) $row`.

**Comment by matej21** [comment]
@dg no it is not

Refactored, thx @JanTvrdik

**Pull request merged**